generated/src/aws-cpp-sdk-s3/include/aws/s3/S3ExpressSigner.h

```diff
@@ -12,63 +12,133 @@
-                const Aws::String &serviceName,
-                const Aws::String &region,
-                PayloadSigningPolicy signingPolicy = PayloadSigningPolicy::RequestDependent,
-                bool urlEscapePath = true,
-                Aws::Auth::AWSSigningAlgorithm signingAlgorithm = Aws::Auth::AWSSigningAlgorithm::SIGV4);
-
-            virtual ~S3ExpressSigner() {};
-
-            const char *GetName() const override;
-
-            bool SignRequest(Aws::Http::HttpRequest &request,
-                const char *region,
-                const char *serviceName,
-                bool signBody
-            ) const override;
+        using SigningFutureOutcome = typename BASECLASS::SigningFutureOutcome;
+        using SigningProperties = typename BASECLASS::SigningProperties;
+        using SigningError = typename BASECLASS::SigningError;
+        explicit S3ExpressSignerBase(const Aws::String& serviceName, const Aws::String& region)
+            : BASECLASS(serviceName, region)
+        {
+        }
+
+        explicit S3ExpressSignerBase(const Aws::String& serviceName, const Aws::String& region, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy policy, bool escapeUrl)
+            : BASECLASS(serviceName, region, policy, escapeUrl)
+        {
+        }
+
 
-            bool PresignRequest(Aws::Http::HttpRequest& request,
+        SigningFutureOutcome sign(std::shared_ptr<Aws::Http::HttpRequest> httpRequest, const smithy::AwsCredentialIdentityBase& identity, SigningProperties properties) override;
+
+        SigningFutureOutcome presign(std::shared_ptr<Aws::Http::HttpRequest> httpRequest, const smithy::AwsCredentialIdentityBase& identity, SigningProperties properties, const Aws::String& region, const Aws::String& serviceName, long long expirationTimeInSeconds) override;
+
+        protected:
+        
+        // Forwarding constructor
+        template <typename... Args>
+        explicit S3ExpressSignerBase(Args&&... args)
+            : BASECLASS(std::forward<Args>(args)...)
+        {
+        }
+
+        inline bool hasRequestId(const Aws::String &requestId) const {
+            std::lock_guard<std::mutex> lock(m_requestProcessing);
+            return m_requestsProcessing.find(requestId) != m_requestsProcessing.end();
+        }
+
+        inline void putRequestId(const Aws::String &requestId) const {
+            std::lock_guard<std::mutex> lock(m_requestProcessing);
+            m_requestsProcessing.insert(requestId);
+        }
+
+        inline void deleteRequestId(const Aws::String &requestId) const {
+            std::lock_guard<std::mutex> lock(m_requestProcessing);
+            m_requestsProcessing.erase(requestId);
+        }
+
+        mutable std::set<Aws::String> m_requestsProcessing;
+        mutable std::mutex m_requestProcessing;
+    };
+
+
+    class AWS_S3_API S3ExpressSigner : public Aws::Client::AWSAuthSigner, public S3ExpressSignerBase<smithy::AwsSigV4Signer>{
```

Comment:
previously `S3ExpressSigner` was defined as

`class AWS_S3_API S3ExpressSigner : public Aws::Client::AWSAuthV4Signer`

which means S3ExpressSigner`is a` Aws::Client::AWSAuthV4Signer. is this public interface change not a breaking change?